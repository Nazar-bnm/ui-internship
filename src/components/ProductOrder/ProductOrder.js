import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import Dropdown from '../shared/Dropdown';
import Button from '../shared/Button';

import './ProductOrder.scss';

export const CN = 'product-order';

const ProductOrder = (props) => {
  const {
    itemName,
    description,
    price,
    sizes,
    colors,
    quantity
  } = props;

  return (
    <div className={`${CN} content`}>
      <h2 className={`${CN}__heading`}>{itemName}</h2>
      <p className={`${CN}__description`}>{description}</p>
      <span className={`${CN}__price`}>{price}</span>
      <div className={`${CN}__dropdowns-wrapper`}>
        <Dropdown options={sizes} />
        <Dropdown options={colors} />
        <Dropdown options={quantity} />
      </div>
      <div className={`${CN}__buttons-wrapper`}>
        <Link to="/not_found">
          <Button customClass={`${CN}__cart-btn`}>add to cart</Button>
        </Link>
        <Link to="/not_found">
          <Button
            customClass={`${CN}__wishlist-btn`}
            icon="heart"
          >
          wishlist
          </Button>
        </Link>
      </div>
    </div>
  );
};

ProductOrder.propTypes = {
  itemName: PropTypes.string,
  description: PropTypes.string,
  price: PropTypes.string,
  quantity: PropTypes.array,
  sizes: PropTypes.array,
  colors: PropTypes.array
};

ProductOrder.defaultProps = {
  itemName: '',
  description: '',
  price: '',
  quantity: [],
  sizes: [],
  colors: []
};

export default ProductOrder;
