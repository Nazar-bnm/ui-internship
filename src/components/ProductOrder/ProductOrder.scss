@import '../../scss/global_styles';

.product-order {
  display: flex;
  flex-direction: column;
  max-width: 300px;
  height: 400px;

  &__heading {
    margin: 10px 0;
    font-size: 2rem;
    font-family: font(secondary);
  }
}