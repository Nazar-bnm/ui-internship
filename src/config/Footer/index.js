export const linkListItems = {
  guide: {
    heading: 'Shopping guide',
    data: [
      { title: 'About G-Star Raw', link: '#' },
      { title: 'Corporate Responsibility', link: '#' },
      { title: 'Press Room', link: '#' },
      { title: 'Careers', link: '#' },
      { title: 'G-Star Retailers', link: '#' },
      { title: 'Open a G-Star Store', link: '#' },
    ],
  },
  help: {
    heading: 'Help & Info',
    data: [
      { title: 'FAQ', link: '#' },
      { title: 'Contact', link: '#' },
      { title: 'Privacy Policy', link: '#' },
      { title: 'Terms & Conditions', link: '#' },
      { title: 'Disclaimer', link: '#' },
    ],
  },
};

export const icons = [
  { name: 'twitter', link: 'https://twitter.com/griddynamics' },
  { name: 'mail', link: 'mailto:khrystynavaskovych@gmail.com' },
  { name: 'facebook', link: 'https://www.facebook.com/griddynamics' },
  { name: 'linkedin', link: 'https://www.linkedin.com/company/grid-dynamics/' },
  { name: 'instagram', link: 'https://www.instagram.com/griddynamics_ua/' },
  {
    name: 'google plus g',
    link: 'https://plus.google.com/115302417170674279390',
  },
];

export const paymentIcons = [
  { name: 'cc paypal', link: '#' },
  { name: 'cc visa', link: '#' },
  { name: 'cc mastercard', link: '#' },
];

export const linkListItemsTablet = [
  { title: 'About G-Star Raw', link: '#' },
  { title: 'G-Star Retailers', link: '#' },
  { title: 'FAQ', link: '#' },
  { title: 'Contact', link: '#' },
  { title: 'Privacy Policy', link: '#' },
  { title: 'Terms & Conditions', link: '#' },
];
