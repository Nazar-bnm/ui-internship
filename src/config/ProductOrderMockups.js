export const productOrderParameters = {
  sizes: [
    { value: 'size:', label: 'size:' },
    { value: 's', label: 's' },
    { value: 'm', label: 'm' },
    { value: 'l', label: 'l' },
    { value: 'xl', label: 'xl' },
  ],
  colors: [
    { value: 'color:', label: 'color:' },
    { value: 'white', label: 'white' },
    { value: 'pink', label: 'pink' },
    { value: 'black', label: 'black' },
  ],
  quantity: [
    { value: 'qty:', label: 'qty:' },
    { value: '1', label: '1' },
    { value: '2', label: '2' },
    { value: '3', label: '3' },
    { value: '4', label: '4' },
    { value: '5', label: '5' },
    { value: '6', label: '6' },
    { value: '7', label: '7' },
    { value: '8', label: '8' },
    { value: '9', label: '9' },
    { value: '10', label: '10' },
  ],
};
