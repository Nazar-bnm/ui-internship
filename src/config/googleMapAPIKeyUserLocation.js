export const googleMapAPIKeyUserLocation = 'AIzaSyBAjVWlquT_qSKufJyzldhchRMOw46WA1I';