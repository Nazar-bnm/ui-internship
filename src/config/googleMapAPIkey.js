export const googleAPIKey = 'AIzaSyAyesbQMyKVVbBgKVi2g6VX7mop2z96jB';
