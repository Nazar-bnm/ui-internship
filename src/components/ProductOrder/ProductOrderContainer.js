import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import { addToWishlist } from '../../actions/actionsWishlist';
import ProductOrder from './ProductOrder';

const mapStateToProps = () => ({});

const mapDispatchToProps = (dispatch) => ({
  addToWishlist: bindActionCreators(addToWishlist, dispatch)
});

export default connect(mapStateToProps, mapDispatchToProps)(ProductOrder);
