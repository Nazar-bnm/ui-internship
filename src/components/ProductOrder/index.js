import ProductOrder from './ProductOrder';

export default ProductOrder;
