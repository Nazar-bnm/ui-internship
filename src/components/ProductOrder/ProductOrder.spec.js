import React from 'react';
import { shallow } from 'enzyme';

import ProductOrder from './ProductOrder';
import { productOrderParameters } from '../../config/ProductOrderMockups';

describe('<ProductOrder />', () => {
  test('should match snapshot', () => {
    const wrapper = shallow(<ProductOrder {...productOrderParameters} />);
    expect(wrapper).toMatchSnapshot();
  });
});
