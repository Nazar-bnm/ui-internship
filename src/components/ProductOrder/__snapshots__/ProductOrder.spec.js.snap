// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductOrder /> should match snapshot 1`] = `
<div
  className="product-order"
>
  <h2
    className="product-order__heading"
  />
  <p
    className="product-order__description"
  />
  <span
    className="product-order__price"
  />
  <div
    className="product-order__dropdowns-wrapper"
  >
    <Dropdown
      className=""
      menuOptions={
        Array [
          Object {
            "label": "size:",
            "value": "size:",
          },
          Object {
            "label": "s",
            "value": "s",
          },
          Object {
            "label": "m",
            "value": "m",
          },
          Object {
            "label": "l",
            "value": "l",
          },
          Object {
            "label": "xl",
            "value": "xl",
          },
        ]
      }
      onDropdownChange={[Function]}
      placeholder="size:"
      value=""
    />
    <Dropdown
      className=""
      menuOptions={
        Array [
          Object {
            "label": "color:",
            "value": "color:",
          },
          Object {
            "label": "white",
            "value": "white",
          },
          Object {
            "label": "pink",
            "value": "pink",
          },
          Object {
            "label": "black",
            "value": "black",
          },
        ]
      }
      onDropdownChange={[Function]}
      placeholder="color:"
      value=""
    />
    <Dropdown
      className=""
      menuOptions={
        Array [
          Object {
            "label": "qty:",
            "value": "qty:",
          },
          Object {
            "label": "1",
            "value": "1",
          },
          Object {
            "label": "2",
            "value": "2",
          },
          Object {
            "label": "3",
            "value": "3",
          },
          Object {
            "label": "4",
            "value": "4",
          },
          Object {
            "label": "5",
            "value": "5",
          },
          Object {
            "label": "6",
            "value": "6",
          },
          Object {
            "label": "7",
            "value": "7",
          },
          Object {
            "label": "8",
            "value": "8",
          },
          Object {
            "label": "9",
            "value": "9",
          },
          Object {
            "label": "10",
            "value": "10",
          },
        ]
      }
      onDropdownChange={[Function]}
      placeholder="qty:"
      value=""
    />
  </div>
  <div
    className="product-order__buttons-wrapper"
  >
    <Link
      to="/cart"
    >
      <Button
        className="product-order__cart-btn"
        customClass={null}
        icon={null}
        onClick={[Function]}
      >
        add to cart
      </Button>
    </Link>
    <Link
      to="/wishlist"
    >
      <Button
        className="product-order__wishlist-btn"
        customClass={null}
        icon="heart"
        onClick={[Function]}
      >
        wishlist
      </Button>
    </Link>
  </div>
</div>
`;
